algebra-core/src/serialize/flags.rs

```diff
@@ -0,0 +1,125 @@
+pub trait Flags: Default + Clone + Copy + Sized {
+    fn u64_bitmask(&self) -> u64;
+    fn from_u64(value: u64) -> Self;
+    fn from_u64_remove_flags(value: &mut u64) -> Self;
+    fn len() -> usize;
+}
+
+/// Flags to be encoded into the serialization.
+/// The default flags (empty) should not change the binary representation.
+#[derive(Default, Clone, Copy)]
+pub struct EmptyFlags;
+
+impl Flags for EmptyFlags {
+    fn u64_bitmask(&self) -> u64 {
+        0
+    }
+
+    fn from_u64(_value: u64) -> Self {
+        EmptyFlags
+    }
+
+    fn from_u64_remove_flags(_value: &mut u64) -> Self {
+        EmptyFlags
+    }
+
+    fn len() -> usize {
+        0
+    }
+}
+
+/// Flags to be encoded into the serialization.
+/// The default flags (empty) should not change the binary representation.
+#[derive(Default, Clone, Copy)]
+pub struct SWFlags {
+    pub y_sign:      bool,
+    pub is_infinity: bool,
+}
```

Comment:
Since the two cases are mutually exclusize, can we make this an enum?
```suggestion
pub enum SWFlags {
    Infinity,
    PositiveY,
    NegativeY,
}
```